website/docs/developer-section/quickstart/oa-embedded-qrcode.md

```diff
@@ -0,0 +1,222 @@
+
+The syntax of the shareable URL is in the following: 
+
+
+```
+https://[VERIFIER_URL]/?q=[ENCODED_QUERY_PARAM]#[OPTIONAL_DECRYPTION_KEY]
+```
+
+
+In the URL, the query parameter (q) consists of an encoded query string of a JSON. If you analyze the URL in your preferred URL decoder, you will see the query parameter (q) decoded to a JSON string similar to the following example:
+
+
+```
+{
+  "type": "DOCUMENT",
+  "payload": {
+    "uri": "https://gallery.openattestation.com/static/documents/transcript-encrypted.opencert",
+    "permittedActions": [
+      "STORE"
+    ],
+    "redirect": "https://dev.opencerts.io"
+  }
+}
+```
+
+
+
+For the ease of reading, here’s how an **unencoded** URL would look like:
+
+
+
+[https://dev.opencerts.io/?q= **{"type":"DOCUMENT","payload":{"uri":"https://gallery.openattestation.com/static/documents/transcript-encrypted.opencert","permittedActions":["STORE"],"redirect":"https://dev.opencerts.io"}}**#**{"key":"691add1930798b63b17c8683a4776bedc16771ea5664337e21a563be0529024f"}**​](https://dev.opencerts.io/?q={"type":"DOCUMENT","payload":{"uri":"https://gallery.openattestation.com/static/documents/transcript-encrypted.opencert","permittedActions":["STORE"],"redirect":"https://dev.opencerts.io"}}#{"key":"691add1930798b63b17c8683a4776bedc16771ea5664337e21a563be0529024f"})
+
+
+
+>**Note:** The bold portion has to be encoded separately.
+
+
+### Mandatory information in the QR code
+
+The table below describes the JSON key/value pairs for the query parameter `q` for the verifier to handle: 
+
+
+<table>
+  <tr>
+   <td><strong>Query Parameter <code>q</code></strong>
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Key</strong>
+   </td>
+   <td><strong>Necessity</strong>
+   </td>
+   <td><strong>Definition</strong>
+   </td>
+   <td><strong>Data Type</strong>
+   </td>
+   <td><strong>Value</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>type</code>
+   </td>
+   <td>Required
+   </td>
+   <td>Data type of the information that the QR code presents.
+   </td>
+   <td>Enum
+   </td>
+   <td>The following value option is accepted:
+<ul>
+
+<li><code>DOCUMENT</code>
+</li>
+</ul>
+   </td>
+  </tr>
+  <tr>
+   <td><code>payload</code>
+   </td>
+   <td>Required
+   </td>
+   <td>An object that consists of three variables, including <code>uri</code>, <code>permittedActions</code>, and <code>redirect</code>.
+   </td>
+   <td>Object
+   </td>
+   <td>The three variables in this object must follow the respective requirements on their value options.
+   </td>
+  </tr>
+  <tr>
+   <td><code>payload.uri</code>
+   </td>
+   <td>Required
+   </td>
+   <td>A URI where the verifiable document is hosted.
+   </td>
+   <td>URI
+   </td>
+   <td>A valid URI pointing to the document that is retrievable by the verifier. 
+<p>Example: S3 Bucket</p>
+   </td>
+  </tr>
+  <tr>
+   <td><code>payload.permittedActions</code>
```

Comment:
After clarifying with Kyle, `payload.permittedActions` and `payload.redirect` are optional fields and are not necessary for the link to work. As of now, only `payload.uri` is required and if the document is encrypted then it will also require the corresponding `key` (which is correctly documented now).

Can we update this table to show that these 2 fields are optional? As of now their functionality is also limited/experimental so for users that want to create QR codes it will be mainly the uri and key portion that is important to know.

Reply:
thanks for clarifying - changed to "optional"

Reply:
We should move this to the section below as this section is titled `Mandatory information in the QR code`

Reply:
Sure! the optional fields are moved into the `Optional information...` table.